src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -4,21 +4,64 @@ Imports System.Collections.Concurrent
 Imports System.Collections.Immutable
 Imports Microsoft.CodeAnalysis.PooledObjects
 Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
 
 Namespace Microsoft.CodeAnalysis.Semantics
     Partial Friend NotInheritable Class VisualBasicOperationFactory
 
         Private ReadOnly _cache As ConcurrentDictionary(Of BoundNode, IOperation) =
             New ConcurrentDictionary(Of BoundNode, IOperation)(concurrencyLevel:=2, capacity:=10)
 
+        Private ReadOnly _semanticModel As SemanticModel
+
+        Public Sub New(semanticModel As SemanticModel)
+            _semanticModel = semanticModel
+        End Sub
+
         Public Function Create(boundNode As BoundNode) As IOperation
             If boundNode Is Nothing Then
                 Return Nothing
             End If
 
+            ' this should be removed once this issue is fixed
+            ' https://github.com/dotnet/roslyn/issues/21186
+            If TypeOf boundNode Is BoundValuePlaceholderBase Then
+                ' since same place holder bound node appears in multiple places in the tree
+                ' we can't use bound node to operation map.
+                ' for now, we will just create new operation and return clone but we need to figure out
+                ' what we want to do with place holder node such as just returning nothing
+                Return CreateInternal(boundNode).Clone()
+            End If
+
+            ' this should be removed once this issue is fixed
+            ' https://github.com/dotnet/roslyn/issues/21187
+            If IsIgnoredNode(boundNode) Then
+                ' due to how IOperation is set up, some of VB BoundNode must be ignored
+                ' while generating IOperation. otherwise, 2 different IOperation trees will be created
+                ' for nodes under same sub tree
+                Return Nothing
+            End If
+
             Return _cache.GetOrAdd(boundNode, Function(n) CreateInternal(n))
         End Function
 
+        Private Function IsIgnoredNode(boundNode As BoundNode) As Boolean
+            ' since boundNode doesn't have parent pointer, it can't just look around using bound node
+            ' it needs to use syntax node
+            If TypeOf boundNode Is BoundExpressionStatement Then
+                Return If(boundNode?.Syntax.Kind() = SyntaxKind.SelectStatement, False)
+            ElseIf TypeOf boundNode Is BoundCaseBlock Then
+                Return True
+            ElseIf TypeOf boundNode Is BoundCaseStatement Then
+                Return True
+            ElseIf TypeOf boundNode Is BoundEventAccess Then
+                Return If(boundNode?.Syntax.Parent.Kind() = SyntaxKind.AddHandlerStatement, False) OrElse
+                       If(boundNode?.Syntax.Parent.Kind() = SyntaxKind.RemoveHandlerStatement, False)
+            End If
```

Comment:
Can the method be written with `Select Case boundNode.Kind` rather than `TypeOf` checks?